Src/FluentAssertions/Collections/MaximumMatching/MaximumMatchingProblem.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using FluentAssertions.Collections.MaximumMatching;
```

Comment:
The algorithm is called [Ford-Fulkerson algorithm](https://en.wikipedia.org/wiki/Ford%E2%80%93Fulkerson_algorithm) and it can be used to solve the Maximum _Flow_ Problem.
The Maximum Matching Problem can be modelled as a Maximum Flow Problem.

We want to find a _perfect_ matching in a bipartite graph, i.e. a one-one-one mapping between all predicates and items.
This can be solved by finding a maximum matching and if no predicates or items are unmatched, it is perfect.

Maximum and maximal are two different things in [matching theory](https://en.wikipedia.org/wiki/Matching_(graph_theory)#Definitions).
In plain words, a _maximal_ matching is some valid solution, where a _maximum_ matching is a solution that is no less worse than any other solution.

Reply:
The algorithm is called [Ford-Fulkerson algorithm](https://en.wikipedia.org/wiki/Ford%E2%80%93Fulkerson_algorithm) and it can be used to solve the Maximum _Flow_ Problem.
The Maximum Matching Problem can be modelled as a Maximum Flow Problem.

We want to find a _perfect_ matching in a bipartite graph, i.e. a one-one-one mapping between all predicates and items.
This can be solved by finding a maximum matching and if no predicates or items are unmatched, it is perfect.

Maximum and maximal are two different things in [matching theory](https://en.wikipedia.org/wiki/Matching_(graph_theory)#Definitions).
In plain words, a _maximal_ matching is some valid solution, where a _maximum_ matching is a solution that is no less worse than any other solution.

Reply:
I learned something new today 😜